src/extensions/qdatabase.c

```diff
@@ -365,6 +456,20 @@ static int execute_update(qdb_t *db, const char *query)
         if ((affected = mysql_affected_rows(db->mysql)) < 0) affected = -1;
     }
 
+    Q_MUTEX_LEAVE(db->qmutex);
+    return affected;
+#elif defined(Q_ENABLE_PGSQL)
+    Q_MUTEX_ENTER(db->qmutex);
+
+    int affected = -1;
+
+    // query
+    DEBUG("%s", query);
+    if (pgsql_query(db->pgsql, query, PQ_WRITE) == 0) {
+        /* get affected rows */
+        if ((affected = pgsql_affected_rows(db->pgsql)) < 0) affected = -1;
+    }
+
     Q_MUTEX_LEAVE(db->qmutex);
     return affected;
 #else
```

Comment:
Maybe we can furder reduce the code duplication like the following example:

#if defined(Q_ENABLE_MYSQL) || defined(Q_ENABLE_PGSQL)

common codes

#ifdef Q_ENABLE_MYSQL
    if (mysql_query(db->mysql, query) == 0) {
        if ((affected = mysql_affected_rows(db->mysql)) < 0) affected = -1;
    }
#elif defined(Q_ENABLE_PGSQL)
    if (pgsql_query(db->pgsql, query, PQ_WRITE) == 0) {
        if ((affected = pgsql_affected_rows(db->pgsql)) < 0) affected = -1;
    }
#endif

#endif